libbpf-tools/klockstat.bpf.c

```diff
@@ -156,15 +161,18 @@ static void account(struct lockholder_info *li)
 	 * the same lock, which is what we want, but we need to use atomics to
 	 * avoid corruption, especially for the total_time variables.
 	 */
-	ls = bpf_map_lookup_elem(&stat_map, &li->stack_id);
+	ls = bpf_map_lookup_elem(&stat_map, &key);
 	if (!ls) {
 		struct lock_stat fresh = {0};
 
-		bpf_map_update_elem(&stat_map, &li->stack_id, &fresh, BPF_ANY);
-		ls = bpf_map_lookup_elem(&stat_map, &li->stack_id);
+		bpf_map_update_elem(&stat_map, &key, &fresh, BPF_ANY);
+		ls = bpf_map_lookup_elem(&stat_map, &key);
 		if (!ls)
 			return;
-	}
+
+		if (per_thread)
+			bpf_get_current_comm(ls->acq_max_comm, TASK_COMM_LEN);
```

Comment:
nice.  much easier to tuck it away in the lock_stat.  we already have the space available, and its probably not worth making a per-thread pid-to-comm map.

Reply:
Right.